pkg/cloudprovider/provider/anexia/provider.go

```diff
@@ -76,10 +79,12 @@ type resolvedDisk struct {
 type resolvedConfig struct {
 	anxtypes.RawConfig
 
-	Token      string
-	VlanID     string
-	LocationID string
-	TemplateID string
+	Token         string
+	VlanID        string
+	LocationID    string
+	TemplateID    string
+	Template      string
+	TemplateBuild string
```

Comment:
Maybe `resolvedConfig` should only contain the `TemplateID`, with the `resolveConfig()` resolving the given `Template` and `TemplateBuild` to the correct `TemplateID`?

Reply:
Unfortunately this isn't as trivial as it appears. `resolveConfig()` would need an api.API instance to resolve the template ID from a named template, but the result of `resolveConfig()` (`token`) is required to instantiate `api.API`. Calling `getClient` inside the function would break mock testing.